tests/cpp-tests/Classes/FileUtilsTest/FileUtilsTest.cpp

```diff
@@ -592,6 +593,127 @@ std::string TestWriteString::subtitle() const
     return "";
 }
 
+class CustomBuffer : public ResizableBuffer {};
+
+struct AlreadyExistsBuffer {};
+
+NS_CC_BEGIN
```

Comment:
It is definitely explicit specialization.

C++ does not support defined a class with `< XX >` eg:

```C++
class A<int> {
};
```


After :

* remove NS_CC_BEGIN NS_CC_END
* and changed to class ResizableBufferAdapter<AlreadyExistsBuffer> : public cocos2d::ResizableBuffer

That will got the follow error on VS2015:

```
1>  FileUtilsTest.cpp
1>..\Classes\FileUtilsTest\FileUtilsTest.cpp(601): error C2906: 'cocos2d::ResizableBufferAdapter<AlreadyExistsBuffer>': explicit specialization requires 'template <>'
```